transform_stats_to_metrics.go

```diff
@@ -237,25 +237,38 @@ func setPointValue(pt *metricspb.Point, value float64, mType measureType) {
 	}
 }
 
-func exemplarsToDistributionBuckets(exemplars []*exemplar.Exemplar) []*metricspb.DistributionValue_Bucket {
-	if len(exemplars) == 0 {
-		return nil
-	}
+// countPerBucket and exemplars are of the same length in well formed data,
+// otherwise ensure that even if exemplars are non-existent that we always
+// insert counts and create distribution value buckets.
+func bucketsToProtoBuckets(countPerBucket []int64, exemplars []*exemplar.Exemplar) []*metricspb.DistributionValue_Bucket {
+	distBuckets := make([]*metricspb.DistributionValue_Bucket, len(countPerBucket))
+	for i := 0; i < len(countPerBucket); i++ {
+		count := countPerBucket[i]
+
+		var exmplr *exemplar.Exemplar
+		if i < len(exemplars) {
+			exmplr = exemplars[i]
+		}
 
-	distBuckets := make([]*metricspb.DistributionValue_Bucket, 0, len(exemplars))
-	for _, exmplr := range exemplars {
-		if exmplr == nil {
+		// In this case no data is present at all, keep the distributionValue bucket nil
+		if count == 0 && exmplr == nil {
+			distBuckets[i] = nil // explicitly using this insertion of nil, for easy readability and debugging
```

Comment:
IIRC if the count is 0, we still put a Bucket proto into the distribution. I believe this is the case for other languages /cc @mayurkale22 @bogdandrutu for confirmation.

Reply:
I just gave this another thought and I agree, even with a nil exempler let's still insert a bucket value with count 0 since for even the legacy case we only had bucket counts without exemplars. Thanks for raising that @songy23!